gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+
+
+def get_positional_arg(args: List[str], pos_arg_name: str) -> Optional[str]:
+    if pos_arg_name in args:
+        return args[args.index(pos_arg_name) + 1]
+
+    return None
+
+
+def append_python_module_to_proc_wd(process: Process, module_name: str) -> str:
+    return process.cwd().replace("/", ".") + module_name
+
+
+class GunicornApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "gunicorn"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "gunicorn"
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app` only in 20.1.0)
+        for arg in process.cmdline():
+            if ":" in arg:
+                if IP_PORT_RE.match(arg):
+                    continue
+
+                return append_python_module_to_proc_wd(process, arg)
+
+        logger.warning(f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module")
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "uwsgi"
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
```

Comment:
Same as in [here](https://github.com/Granulate/gprofiler/pull/242/files#r765956354)

Reply:
Done, resolving.